src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -443,10 +443,56 @@ object Parser {
           val pres = clauses.collect{ case x: RequiresClause => x.exp }
           val preserves = clauses.collect{ case x: PreservesClause => x.exp }
           val posts = clauses.collect{ case x: EnsuresClause => x.exp }
+          val terminationMeasure = {
+            val t = clauses.collect{ case x: DecreasesClause => x.measure}
+            if(t.size <= 1){
+              t.headOption
+            } else {
+              Violation.violation("Unexpected amount of decreases clause")
+            }
+          }
           val isPure = pure.nonEmpty || clauses.contains(PureClause)
-          PFunctionSpec(pres, preserves, posts, isPure)
+          PFunctionSpec(pres, preserves, posts, terminationMeasure, isPure)
       }
     }
+    
+    lazy val measures:Parser[PTerminationMeasure]=
+    "*"  ^^^ PStarCharacter() | ("_" <~ eos) ^^^ PUnderscoreCharacter() | repsep(expression,",") <~ eos ^^ PTupleTerminationMeasure | firstConditionalMeasure 
+
+    lazy val firstConditionalMeasure: Parser[PConditionalMeasureCollection]=
```

Comment:
This is hard to read and understand. Please reformat the code and, if necessary, split this into multiple methods to make it easier to understand. Also, it seems to me that much of the complexity of parsing termination measures come from the fact that the definition of `PTerminationMeasure` is not as simple as it could be, as mentioned in a previous comment.